### ISSUE TITLE
weird parsed error: `std::pair<int,std::pair<int,int>>，std::pair<std::string, std::pair<int,int>>`

### ISSUE THREAD

**Issue by ericxsun** [bytedeco/javacpp-presets#614]
pom of chilitags is not changed. (version: `javacpp-presets 1.4.3-SNAPSHOT`)

I added two simple lines in  `chilitags/src/main/java/org/bytedeco/javacpp/presets/chilitags.java`, 
```
.put(new Info("std::pair<int,int>").pointerTypes("IntIntPair").define())
.put(new Info("std::pair<int,std::pair<int,int> >").pointerTypes("IntIntPairIntPair").define())
```

while compiling with `mvn clean install --projects .,chilitags` , weird errors occurred
first one:
```
JNIEXPORT jobject JNICALL Java_org_bytedeco_javacpp_chilitags_00024IntIntPairIntPair_second__I(JNIEnv* env, jobject obj, jint arg0) {
    ::std::pair<int,std::pair<int,int> >* ptr = (::std::pair<int,std::pair<int,int> >*)jlong_to_ptr(env->GetLongField(obj, JavaCPP_addressFID));
    if (ptr == NULL) {
        env->ThrowNew(JavaCPP_getClass(env, 8), "This pointer address is NULL.");
        return 0;
    }
    jlong position = env->GetLongField(obj, JavaCPP_positionFID);
    ptr += position;
    jobject rarg = obj;
    ptr->second = arg0;
    return rarg;
}
```

```
/chilitags/target/classes/org/bytedeco/javacpp/jnichilitags.cpp:2375:17: error: no viable overloaded '='
    ptr->second = arg0;
    ~~~~~~~~~~~ ^ ~~~~
```

second one:
```
JNIEXPORT jint JNICALL Java_org_bytedeco_javacpp_chilitags_00024IntIntPairIntPair_second__(JNIEnv* env, jobject obj) {
    ::std::pair<int,std::pair<int,int> >* ptr = (::std::pair<int,std::pair<int,int> >*)jlong_to_ptr(env->GetLongField(obj, JavaCPP_addressFID));
    if (ptr == NULL) {
        env->ThrowNew(JavaCPP_getClass(env, 8), "This pointer address is NULL.");
        return 0;
    }
    jlong position = env->GetLongField(obj, JavaCPP_positionFID);
    ptr += position;
    jint rarg = 0;
    int rval = ptr->second;
    rarg = (jint)rval;
    return rarg;
}
```

```
chilitags/target/classes/org/bytedeco/javacpp/jnichilitags.cpp:2387:9: error: no viable conversion from 'std::__1::pair<int, int>' to 'int'
    int rval = ptr->second;
        ^      ~~~~~~~~~~~
```

type of ptr's second value is `std::pair<int, int>`, why that was `int <---> std::pair<int, int>` conversion.

Thanks a lot for your help.

**Comment by ericxsun** [comment]
When only adding 
```
.put(new Info("std::pair<int,int>").pointerTypes("IntIntPair").define())
.put(new Info("std::pair<string,std::pair<int,int> >").pointerTypes("IntIntPairStringPair").define())
```

such error will occurred
```
JNIEXPORT jint JNICALL Java_org_bytedeco_javacpp_chilitags_00024IntIntPairStringPair_first__(JNIEnv* env, jobject obj) {
    ::std::pair<std::string,std::pair<int,int> >* ptr = (::std::pair<std::string,std::pair<int,int> >*)jlong_to_ptr(env->GetLongField(obj, JavaCPP_addressFID));
    if (ptr == NULL) {
        env->ThrowNew(JavaCPP_getClass(env, 8), "This pointer address is NULL.");
        return 0;
    }
    jlong position = env->GetLongField(obj, JavaCPP_positionFID);
    ptr += position;
    jint rarg = 0;
    int rval = ptr->first;
    rarg = (jint)rval;
    return rarg;
}
```

```
chilitags/target/classes/org/bytedeco/javacpp/jnichilitags.cpp:2336:9: error: no viable conversion from 'std::__1::basic_string<char>' to 'int'
    int rval = ptr->first;
        ^      ~~~~~~~~~~
```

**Comment by ericxsun** [comment]
@saudet Is there any plan for fixing it?

**Comment by saudet** [comment]
Yes, I'll look into that shortly. Thanks for reporting!

**Comment by saudet** [comment]
I've fixed that with commit https://github.com/bytedeco/javacpp/commit/2f4e1d22b7bd1b4bfa164c51804de54731d8a0b2, thanks for reporting!

**Comment by ericxsun** [comment]
@saudet Thank you a lot.

**Comment by saudet** [comment]
Fix included in JavaCPP 1.4.3! Thanks again for reporting.
